wiki/TypeScript-Migration.md

```diff
@@ -0,0 +1,48 @@
+# TypeScript migration
+
+The following steps are recommended for migrating components to TypeScript.
+
+1. [Pick a component](https://hashicorp.atlassian.net/browse/HDS-2392), assign yourself to the task in Jira, and add it to the current sprint
+
+    - [This spreadsheet](https://docs.google.com/spreadsheets/d/1sWzLSP8TUb3WYibYNZ5w6UfEb338DQbZkVaTjK8b1tw/edit#gid=0) contains (more or less) all the dependencies between components, so it’s easier to understand what is ready to be converted (e.g. has dependencies that are already converted to TypeSscript) and what is not.
+
+2. Rename the backing class from `.js` to `.ts`; commit after this step to make it easier to review the changes
+
+3. Opening the files (`.ts` and `.hbs`) in VS Code should show you the linting errors (if you have the [Glint extension](https://marketplace.visualstudio.com/items?itemName=typed-ember.glint-vscode)); similarly, if you are running `yarn lint` in `packages/components`
+
+4. Define component signature
+
+    - You can either do it manually [following the documentation](https://typed-ember.gitbook.io/glint/environments/ember/component-signatures#glimmer-components)
+    - Or, you can generate it using `npx ember-codemod-args-to-signature --src /path/to/component`. This codemod renames the component class name to `IndexComponent` by default, so make sure to revert it (e.g. `HdsBadgeIndexComponent`)
```

Comment:
Should this be:
(e.g. `HdsBadgeComponent`)

instead of:
(e.g. `HdsBadgeIndexComponent`)

?

Reply:
@hashicorp/hds-engineering I think this is something we have to figure out/decide: I _think_ I've seen both (with `Index` and without `Index`). Which one do we prefer? If we follow the previous convention, it's without `Index`.

Reply:
Update: We discussed and decided upon not including "Index" as part of the naming at all.